src/ResourceManager/KeyVault/Commands.KeyVault/Commands/ManagedStorageAccounts/NewAzureKeyVaultManagedStorageAccountKey.cs

```diff
@@ -0,0 +1,87 @@
+﻿// ----------------------------------------------------------------------------------
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.KeyVault.Models;
+using System.Globalization;
+using System.Management.Automation;
+using KeyVaultProperties = Microsoft.Azure.Commands.KeyVault.Properties;
+
+namespace Microsoft.Azure.Commands.KeyVault
+{
+    [Cmdlet( VerbsCommon.New, CmdletNoun.AzureKeyVaultManagedStorageAccountKey,
+        SupportsShouldProcess = true,
+         ConfirmImpact = ConfirmImpact.High,
+        HelpUri = Constants.KeyVaultHelpUri )]
+    [OutputType( typeof( ManagedStorageAccount ) )]
+    public class NewAzureKeyVaultManagedStorageAccountKey : KeyVaultCmdletBase
+    {
+        #region Input Parameter Definitions
+        [Parameter( Mandatory = true,
+            Position = 0,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Vault name. Cmdlet constructs the FQDN of a vault based on the name and currently selected environment." )]
+        [ValidateNotNullOrEmpty]
+        public string VaultName { get; set; }
+
+        [Parameter( Mandatory = true,
+            Position = 1,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Key Vault managed storage account name. Cmdlet constructs the FQDN of a managed storage account name from vault name, currently " +
+                          "selected environment and manged storage account name." )]
+        [ValidateNotNullOrEmpty]
+        [Alias( Constants.StorageAccountName )]
+        public string AccountName { get; set; }
+
+        [Parameter( Mandatory = true,
+            Position = 2,
+            HelpMessage = "Name of storage account key to regenerate and make active." )]
+        [ValidateNotNullOrEmpty]
+        public string KeyName { get; set; }
+
+        /// <summary>
+        /// If present, do not ask for confirmation
+        /// </summary>
+        [Parameter( Mandatory = false, HelpMessage = "Do not ask for confirmation." )]
```

Comment:
No, that is precisely the point.  Asking for confirmation on every invocation does not provide the user with any useful information about the effects of that invocation.  Therefore, you should just implement ShouldProcess and *not* have a Force parameter.